resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ResourceMethodRegistry.java

```diff
@@ -366,6 +366,10 @@ protected void processMethod(ResourceFactory rf, String base, ResourceLocator me
             rootNode.addInvoker(fullpath, locator);
          else root.addInvoker(classExpression, fullpath, locator);
       }
+
+      if (rf instanceof SingletonResource) {

```

Comment:
Hi @rnetuka,

I'm questioning whether this should be in
```
   protected void register(ResourceFactory rf, String base, ResourceClass resourceClass)
   {
      for (ResourceMethod method : resourceClass.getResourceMethods())
      {
         processMethod(rf, base, method);
      }
      for (ResourceLocator method : resourceClass.getResourceLocators())
      {
         processMethod(rf, base, method);
      }
   }
```
Otherwise, you'll be calling registerSingletonResource() for each method, no?

-Ron